### DISCUSSION TITLE
How to add values to List<int> ?

### DISCUSSION THREAD

**Discussion by NesurChen** [vfsfitvnm/frida-il2cpp-bridge#371] [Q&A]
Il2Cpp.perform(() => {
        var SystemInt32 = Il2Cpp.domain.assembly("mscorlib.dll").image.class("System.Int32");
        var SCGList = Il2Cpp.domain.assembly("mscorlib.dll").image.class("System.Collections.Generic.List`1");
        var List_I32 = SCGList.inflate(SystemInt32);
        var AddMethod = List_I32.method("Add");
        var List = Il2Cpp.api.objectNew(List_I32);
        AddMethod.invokeRaw(List, 1);
    })
    I wrote as thie. But it doesn't work and shows `Error: abort was called`. 
    How can I add values to the List?

**Comment by vfsfitvnm** [comment]
Hi, I don't really know why you are using low-level methods such as `Il2Cpp.Method::invokeRow` and `Il2Cpp.api::objectNew`, however, the following works just fine:
```ts
Il2Cpp.perform(() => {
    const List = Il2Cpp.corlib.class("System.Collections.Generic.List`1");
    const Int32List = List.inflate(Il2Cpp.corlib.class("System.Int32"));

    const list = Int32List.new();
    list.method("Add").invoke(1);
});
```
Alternatively, you must call `Il2Cpp.api.objectInitialize` first (it calls the object contructor):
```ts
    const list = Il2Cpp.api.objectNew(Int32List);
    Il2Cpp.api.objectInitialize(list, NULL);

    Int32List.method("Add").invokeRaw(list, 1);
```

**Reply by NesurChen** [reply]
Thanks very much. I knew this repo for a long time, but I recently start to use it and have few exp
in using this. I just followed the examples in the source code.

**Comment by etmltg69** [comment]
> Hi, I don't really know why you are using low-level methods such as `Il2Cpp.Method::invokeRow` and `Il2Cpp.api::objectNew`, however, the following works just fine:
> 
> ```ts
> Il2Cpp.perform(() => {
>     const List = Il2Cpp.corlib.class("System.Collections.Generic.List`1");
>     const Int32List = List.inflate(Il2Cpp.corlib.class("System.Int32"));
> 
>     const list = Int32List.new();
>     list.method("Add").invoke(1);
> });
> ```
> 
> Alternatively, you must call `Il2Cpp.api.objectInitialize` first (it calls the object contructor):
> 
> ```ts
>     const list = Il2Cpp.api.objectNew(Int32List);
>     Il2Cpp.api.objectInitialize(list, NULL);
> 
>     Int32List.method("Add").invokeRaw(list, 1);
> ```

I get always "access violation at 0x01" with version 16.2.1 trying this, do you know why this happens?

**Comment by soletsabuse** [comment]
> Hi, I don't really know why you are using low-level methods such as `Il2Cpp.Method::invokeRow` and `Il2Cpp.api::objectNew`, however, the following works just fine:
> 
> ```ts
> Il2Cpp.perform(() => {
>     const List = Il2Cpp.corlib.class("System.Collections.Generic.List`1");
>     const Int32List = List.inflate(Il2Cpp.corlib.class("System.Int32"));
> 
>     const list = Int32List.new();
>     list.method("Add").invoke(1);
> });
> ```
> 
> Alternatively, you must call `Il2Cpp.api.objectInitialize` first (it calls the object contructor):
> 
> ```ts
>     const list = Il2Cpp.api.objectNew(Int32List);
>     Il2Cpp.api.objectInitialize(list, NULL);
> 
>     Int32List.method("Add").invokeRaw(list, 1);
> ```

That's doesn't work also. I think it treats 1 as address. Help us guys.
